### PR TITLE
Bring build files up to standard

### PR DESCRIPTION
**Pull request by cblecker**
Copy Dockerfile, standard.mk, and hack/codecov.sh from managed-velero-operator

This updates us to UBI8, and brings in the framework for codecov

Follow up to #76

### CODE REVIEW

File: standard.mk [hunk 1]
```diff
@@ -80,36 +68,31 @@ gocheck: ## Lint code
 	go vet ./cmd/... ./pkg/...
 
 .PHONY: gobuild
-gobuild: gocheck gotest
```
Comment (jewzaam) [inline standard.mk:83]: Why remove lint and unit tests from build?

File: standard.mk [hunk 2]
```diff
@@ -41,33 +41,21 @@ default: gobuild
...
-.PHONY: skopeo-push
-skopeo-push: container-build
-	skopeo copy \
-		--dest-creds "${QUAY_USER}:${QUAY_TOKEN}" \
-		"docker-daemon:${OPERATOR_IMAGE_URI_LATEST}" \
-		"docker://${OPERATOR_IMAGE_URI_LATEST}"
-	skopeo copy \
-		--dest-creds "${QUAY_USER}:${QUAY_TOKEN}" \
-		"docker-daemon:${OPERATOR_IMAGE_URI}" \
-		"docker://${OPERATOR_IMAGE_URI}"
```
Comment (jewzaam) [inline standard.mk:70]: This is referenced in build/deploy.sh

File: standard.mk (hunk 1 above)
Comment (cblecker) [reply standard.mk:83]: They're run as separate steps, aren't they? We don't need to run them twice.

File: standard.mk (hunk 2 above)
Comment (cblecker) [reply standard.mk:70]: You're right. Bleh. Yeah, we have separate build targets in different repos. Will add these back.

File: standard.mk (hunk 1 above)
Comment (cblecker) [reply standard.mk:83]: Nope, they aren't (yet). Adding this back

File: standard.mk [hunk 3]
```diff
@@ -18,40 +18,37 @@ endif
...
-CONTAINER_ENGINE=$(shell which podman 2>/dev/null || which docker 2>/dev/null)
+CONTAINER_ENGINE=$(shell command -v podman 2>/dev/null || command -v docker 2>/dev/null)
 
-TESTTARGETS := $(shell go list -e ./... | egrep -v "/(vendor)/")
 # ex, -v
 TESTOPTS :=
 
 ALLOW_DIRTY_CHECKOUT?=false
 
-default: gobuild
```
Comment (jewzaam) [inline standard.mk:40]: When I build with `make` locally it's running the "clean" target.  It don't expect it to, but it is.  I wonder if it's defaulting to the first defined target if it hasn't seen a "default" first?  The file `standard.mk` is loaded before "default" is defined in `Makefile`.  Need to add this back in for expected behavior.

**Comment by jewzaam** [comment]
`make container-build` also fails ultimately.  Here's the last step where it hangs:

```
STEP 7: RUN make gobuild
build gopkg.in/yaml.v2: cannot load gopkg.in/yaml.v2: open /workdir/vendor/gopkg.in/yaml.v2: no such file or directory
build gopkg.in/yaml.v2: cannot load gopkg.in/yaml.v2: open /workdir/vendor/gopkg.in/yaml.v2: no such file or directory
build k8s.io/apimachinery/pkg/runtime: cannot load k8s.io/apimachinery/pkg/runtime: open /workdir/vendor/k8s.io/apimachinery/pkg/runtime: no such file or directory
build sigs.k8s.io/controller-runtime/pkg/manager: cannot load sigs.k8s.io/controller-runtime/pkg/manager: open /workdir/vendor/sigs.k8s.io/controller-runtime/pkg/manager: no such file or dire
ctory
build github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1: cannot load github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1: open /workdir/vendor/github.com/coreos/prometheus
-operator/pkg/apis/monitoring/v1: no such file or directory
gofmt -s -l  | grep ".*\.go"; if [ "$?" = "0" ]; then gofmt -s -d ; exit 1; fi
```

I would expect this to work since it's being built in a container.  It's the target used by the PR check which has passed so... local env bleeding into the build container?

Checking ci.ext and I see the same gofmt command but it is successful.  Something is off here.

```
16:00:39 Step 7/15 : RUN make gobuild
16:00:39  ---> Running in 6aaf76a3acd8
16:00:40 build k8s.io/apimachinery/pkg/runtime: cannot load k8s.io/apimachinery/pkg/runtime: open /workdir/vendor/k8s.io/apimachinery/pkg/runtime: no such file or directory
16:00:40 build k8s.io/apimachinery/pkg/runtime: cannot load k8s.io/apimachinery/pkg/runtime: open /workdir/vendor/k8s.io/apimachinery/pkg/runtime: no such file or directory
16:00:40 gofmt -s -l  | grep ".*\.go"; if [ "$?" = "0" ]; then gofmt -s -d ; exit 1; fi
16:00:40 go vet ./cmd/... ./pkg/...
16:01:44 go test  ./...
16:01:58 ?   	github.com/openshift/configure-alertmanager-operator/cmd/manager	[no test files]
16:01:58 ?   	github.com/openshift/configure-alertmanager-operator/config	[no test files]
16:01:58 ?   	github.com/openshift/configure-alertmanager-operator/pkg/apis	[no test files]
16:01:58 ?   	github.com/openshift/configure-alertmanager-operator/pkg/controller	[no test files]
16:02:00 ok  	github.com/openshift/configure-alertmanager-operator/pkg/controller/secret	0.046s
16:02:00 ?   	github.com/openshift/configure-alertmanager-operator/pkg/metrics	[no test files]
16:02:00 ?   	github.com/openshift/configure-alertmanager-operator/pkg/types	[no test files]
16:02:00 ?   	github.com/openshift/configure-alertmanager-operator/version	[no test files]
16:02:00 GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -gcflags="all=-trimpath=/go" -asmflags="all=-trimpath=/go" -o build/_output/bin/configure-alertmanager-operator ./cmd/manager
16:03:26 Removing intermediate container 6aaf76a3acd8
16:03:26  ---> 11596bb0be8c
16:03:26 Step 8/15 : FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
```

**Comment by jewzaam** [comment]
Local issues still but that's my problem... `make container-build` now works.

/lgtm

**Pull request merged**